lib/telemetry.go

```diff
@@ -390,5 +403,45 @@ func InitTelemetry(cfg TelemetryConfig) (*metrics.InmemSink, error) {
 		metricsConf.EnableHostname = false
 		metrics.NewGlobal(metricsConf, memSink)
 	}
+
+	fmt.Fprintln(os.Stderr, "jaeger url", cfg.JaegerURL)
+
+	if cfg.JaegerURL != "" {
+		fmt.Fprintln(os.Stderr, "configuring jaeger exporter and trace provider", cfg.JaegerURL)
+		exp, err := newExporter(context.Background(), cfg.JaegerURL)
+		if err != nil {
+			return nil, err
+		}
+		newTraceProvider(exp)
+	}
+
 	return memSink, nil
 }
+
+func newExporter(ctx context.Context, url string) (*jaeger.Exporter, error) {
+	// Your preferred exporter: console, jaeger, zipkin, OTLP, etc.
+	exporter, err := jaeger.New(jaeger.WithCollectorEndpoint(jaeger.WithEndpoint(url)))
+	return exporter, err
+}
+
+func newTraceProvider(exp sdktrace.SpanExporter) {
+	// The service.name attribute is required.
+	resource, err := resource.New(context.Background(),
+		resource.WithAttributes(semconv.ServiceNameKey.String("consul"), semconv.ServiceVersionKey.String(version.GetHumanVersion())),
+		resource.WithHost(),
+		resource.WithOS(),
+	)
+
+	if err != nil {
+		panic(err)
+	}
+
+	otel.SetTracerProvider(sdktrace.NewTracerProvider(
+		sdktrace.WithBatcher(exp),
+		sdktrace.WithResource(resource),
+		sdktrace.WithSampler(sdktrace.AlwaysSample()),
+	))
+	propagator := propagation.NewCompositeTextMapPropagator(propagation.Baggage{}, propagation.TraceContext{})
```

Comment:
Baggage allows us to propagate _attributes_ between process boundaries but we're not doing that here. Also the TraceContext